lightning/src/chain/channelmonitor.rs

```diff
@@ -2665,12 +2662,32 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		let per_commitment_key = ignore_error!(SecretKey::from_slice(&secret));
 		let per_commitment_point = PublicKey::from_secret_key(&self.secp_ctx, &per_commitment_key);
 
-		log_error!(logger, "Got broadcast of revoked counterparty HTLC transaction, spending {}:{}", htlc_txid, 0);
-		let revk_outp = RevokedOutput::build(per_commitment_point, self.counterparty_commitment_params.counterparty_delayed_payment_base_key, self.counterparty_commitment_params.counterparty_htlc_base_key, per_commitment_key, tx.output[0].value, self.counterparty_commitment_params.on_counterparty_tx_csv);
-		let justice_package = PackageTemplate::build_package(htlc_txid, 0, PackageSolvingData::RevokedOutput(revk_outp), height + self.counterparty_commitment_params.on_counterparty_tx_csv as u32, true, height);
-		let claimable_outpoints = vec!(justice_package);
-		let outputs = vec![(0, tx.output[0].clone())];
-		(claimable_outpoints, Some((htlc_txid, outputs)))
+		let htlc_txid = tx.txid();
+		let mut claimable_outpoints = vec![];
+		let mut outputs_to_watch = None;
+		for (idx, input) in tx.input.iter().enumerate() {
+			// HTLC transactions always spend an output on the commitment transaction with a witness
+			// of 5 elements. The HTLC input will always have a corresponding output at the same
+			// index within the transaction.
+			if input.previous_output.txid == *commitment_txid && input.witness.len() == 5 && tx.output.get(idx).is_some() {
```

Comment:
So the purpose of this logic is to detect the claim of revoked HTLC-timeout or HTLC-success against the counterparty commitment transaction itself. Assuming our own `revocation_basepoint_secret` didn't leak, the only satisfying witness that our counterparty can generate are respectively a `HTLCClaim::OfferedTimeout` or a `HTLCClaim::AcceptedPreimage`. Both of them have a witness of 5 elements. So I think this change should fix the "post-anchor" blinding case raised in #1860. Thanks for that, if you can still a reference to https://lists.linuxfoundation.org/pipermail/lightning-dev/2022-April/003561.html for future-proofing hopefully our code against hazardous refactoring?